pycloudlib/cloud.py

```diff
@@ -35,11 +35,15 @@ def __init__(
 
         user = getpass.getuser()
         self.key_pair = KeyPair(
-            public_key_path=os.path.expanduser(
-                self.config.get("public_key_path", f"~{user}/.ssh/id_rsa.pub")
+            public_key_path=os.path.expandvars(
+                os.path.expanduser(
+                    self.config.get(
+                        "public_key_path", f"~{user}/.ssh/id_rsa.pub"
```

Comment:
Could this default just be ${USER} instead of ${user} to get it from  default environ variables? Or are you looking to actually provide a separate "user" env var in CI that will be different than "jenkins"?

Running this on our jenkins node as the "jenkins" user gives me the following:

```bash
$ python3 -c 'import os; print(os.path.expandvars("test/${USER}"))'
test/jenkins
```

```suggestion
                        "public_key_path", f"~{USER}/.ssh/id_rsa.pub"
```

Reply:
This isn't ${user}, it's an f-string referencing a `user` variable defined about 5 lines up. I don't think the expansion is useful in the default value here...I think it's intended for `public_key_path` possibly containing an env var.

Reply:
We could use `$HOME` (expanded by expandvars) instead of `~{user}` (f-string then expanded by expanduser) now, but we still need `user` (Python variable) a few lines below, in:
```
name=self.config.get("key_name", user),
```
so we can't fully drop it.

Reply:
Right, sorry for the miss on f-string here.  +1.